### ISSUE TITLE
[🐛 Bug]: DevTools timeout is hardcoded and not sufficient for some commands

### ISSUE THREAD

**Issue by dennisoelkers** [SeleniumHQ/selenium#14912]
### What happened?

(If this is a bug or a feature is certainly discussable. I decided to file it as a bug, because it is effectively blocking functionality in our code that should work without having to add a feature)

The timeout used for executing DevTools commands in the Java library is [hardcoded](https://github.com/SeleniumHQ/selenium/blob/trunk/java/src/org/openqa/selenium/devtools/DevTools.java#L45) to 10 seconds. For some commands (e.g. printing a large/complex page to PDF) or on overloaded systems, this might not be enough. It should be configurable.

### How can we reproduce the issue?

```shell
-
```

### Relevant log output

```shell
-
```

### Operating System

Linux

### Selenium version

Java 4.22.0

### What are the browser(s) and version(s) where you see this issue?

Chrome 124

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver 124.0.6367.201

**Comment by navin772** [comment]
Hi @dennisoelkers, thanks for reporting this issue. I think this is a valid request, I will make the required changes and open a PR!

**Comment by navin772** [comment]
@dennisoelkers can you provide the reproducible code you were having issues with so I can test the changes?

**Comment by navin772** [comment]
Hi @dennisoelkers, as per https://github.com/SeleniumHQ/selenium/pull/14931#pullrequestreview-2539839696, we are moving to BiDi completely and CDP functionality will be ~removed~ depreciated.

But, we can increase the hardcoded timeout value. Do you have an approx timeout value that will suffice your tests?

**Comment by diemol** [comment]
To add to the comment above, CDP will not be removed immediately; there will be an extended migration period.

**Comment by navin772** [comment]
Yeah, I meant to say depreciated, thanks for correcting!

**Comment by dennisoelkers** [comment]
Hey @navin772,

> But, we can increase the hardcoded timeout value. Do you have an approx timeout value that will suffice your tests?

We are currently using 180s - it is very conservative (as in too high), but has served us well so far.

In our case, we are using CDP to print a page to PDF. Is it possible to do this using BiDi already? How is BiDi handling the timeout for operations like this, which might take longer?

**Comment by navin772** [comment]
@dennisoelkers we were thinking of increasing it to 30s max.

For BiDi, I think this is the documentation - https://www.selenium.dev/documentation/webdriver/bidi/w3c/browsing_context/#print-page. Implementation [here](https://github.com/SeleniumHQ/selenium/blob/164bf7944b6f7a343de792a5ee48f694286dac3c/java/src/org/openqa/selenium/bidi/browsingcontext/BrowsingContext.java#L340).

@diemol can you confirm if the timeout is configurable in BiDi?

**Comment by diemol** [comment]
I don't think so. If we want to make a change, that is where we should spend time.

@pujagani, do you know?

**Comment by pujagani** [comment]
I do not think the timeout is configurable in BiDi. Currently, It is set to 30 seconds. I think we should make it configurable so end users have that flexibility.

**Comment by dennisoelkers** [comment]
@navin772, @diemol: Would it be so hard to add a configurable timeout to CDP? What we did locally to fix the issue we were seeing with the timeout being too low, was that we monkey-patched the `DevTools` class and added a `sendWithTimeout(Command<V>, Duration)` method. 

We want to get rid of the monkey-patching part, but the interface is quite good for us, because we want to use it for certain feature (e.g. the print command), but for others the default timeout is okay. Would that be an option?

**Comment by diemol** [comment]
@dennisoelkers would you like to send a PR for that?

**Comment by dennisoelkers** [comment]
@diemol: Sure, here it is: https://github.com/SeleniumHQ/selenium/pull/15059
